### PR TITLE
Version asset check

### PR DESCRIPTION
**Pull request by jmatosp**
This PR enable ability to check latest version asset available in version command.

Also temporary the update command displays all available versions..

The hard coded repository URI is going to be replaced on next PR

### CODE REVIEW

File: client/cmd/kcoin/misccmd.go [hunk 1]
```diff
@@ -31,9 +42,17 @@ The output of this command is supposed to be machine-readable.
 	}
 )
 
-func version(ctx *cli.Context) error {
+func versionPrint(ctx *cli.Context) error {
 	fmt.Println(strings.Title(clientIdentifier))
 	fmt.Println("Version:", params.Version)
+
+	// print latest version if available
+	finder := version.NewFinder("https://s3.amazonaws.com/releases.kowala.io")
```
Comment (yourheropaul) [inline client/cmd/kcoin/misccmd.go]: Can these URLs be specified in the CLI command? We can't centralise the location, and that URL will probably change.

File: client/cmd/kcoin/misccmd.go [hunk 2]
```diff
@@ -63,3 +82,18 @@ along with knode. If not, see <http://www.gnu.org/licenses/>.
 `)
 	return nil
 }
+
+func latest(_ *cli.Context) error {
+	finder := version.NewFinder("https://s3.amazonaws.com/releases.kowala.io")
```
Comment (yourheropaul) [inline client/cmd/kcoin/misccmd.go]: The same with this URL -- it needs to be specified on the command line.

**Comment by jmatosp** [comment]
output of `update`

```
$ client/build/bin/kcoin update
Assets Found:
Version: 1.0.10, os: linux, Architeture: amd64, File: kcoin-1.0.10-linux-amd64.zip
Version: 1.0.10, os: linux, Architeture: arm64, File: kcoin-1.0.10-linux-arm64.zip
Version: 1.0.10, os: osx-10.6, Architeture: amd64, File: kcoin-1.0.10-osx-10.6-amd64.zip
Version: 1.0.10, os: windows-4.0, Architeture: amd64, File: kcoin-1.0.10-windows-4.0-amd64.exe.zip
Version: 1.0.11, os: discovery-fix-linux, Architeture: amd64, File: kcoin-1.0.11-discovery-fix-linux-amd64.zip
Version: 1.0.11, os: discovery-fix-linux, Architeture: arm64, File: kcoin-1.0.11-discovery-fix-linux-arm64.zip
Version: 1.0.11, os: discovery-fix-osx-10.6, Architeture: amd64, File: kcoin-1.0.11-discovery-fix-osx-10.6-amd64.zip
Version: 1.0.11, os: discovery-fix-windows-4.0, Architeture: amd64, File: kcoin-1.0.11-discovery-fix-windows-4.0-amd64.exe.zip
Version: 1.0.11, os: improved-docker-build-linux, Architeture: amd64, File: kcoin-1.0.11-improved-docker-build-linux-amd64.zip
Version: 1.0.11, os: improved-docker-build-linux, Architeture: arm64, File: kcoin-1.0.11-improved-docker-build-linux-arm64.zip
Version: 1.0.11, os: improved-docker-build-osx-10.6, Architeture: amd64, File: kcoin-1.0.11-improved-docker-build-osx-10.6-amd64.zip
Version: 1.0.11, os: improved-docker-build-windows-4.0, Architeture: amd64, File: kcoin-1.0.11-improved-docker-build-windows-4.0-amd64.exe.zip
Version: 1.0.14, os: linux, Architeture: amd64, File: kcoin-1.0.14-linux-amd64.zip
Version: 1.0.14, os: linux, Architeture: arm64, File: kcoin-1.0.14-linux-arm64.zip
Version: 1.0.14, os: osx-10.6, Architeture: amd64, File: kcoin-1.0.14-osx-10.6-amd64.zip
Version: 1.0.14, os: windows-4.0, Architeture: amd64, File: kcoin-1.0.14-windows-4.0-amd64.exe.zip
```

**Comment by jmatosp** [comment]
output of `version`

```
$ client/build/bin/kcoin version
Kcoin
Version: 1.0.0-unstable
Latest Version Available: 1.0.14
BuildTime: 2018-08-02T10:03:25Z
Architecture: amd64
Constants Versions: [1]
Go Version: go1.10
Operating System: linux
```

File: Makefile [hunk 3]
```diff
@@ -123,6 +123,10 @@ wallet_backend_dep: dep
 
 # Cross Compilation Targets (xgo)
 
+.PHONY: repository_index
+repository_index:
+	@aws s3 ls releases.kowala.io | cut -b32- - > index.txt
```
Comment (yourheropaul) [inline Makefile]: That's clever. It's a shame the aws command doesn't support structured output, but it will do.

Comment (jmatosp) [reply Makefile]: just a temp helper to generate the file, will sort out later on CI

File: client/cmd/kcoin/misccmd.go (hunk 1 above)
Comment (jmatosp) [reply client/cmd/kcoin/misccmd.go]: yeah, I commented this is temp, but I can change on this PR

**Pull request merged**